x/schema/keeper/schema.go

```diff
@@ -1,14 +1,62 @@
 package keeper
 
 import (
+	"bytes"
 	"encoding/binary"
+	"encoding/json"
+	"os"
+	"path/filepath"
 
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/google/uuid"
+	shell "github.com/ipfs/go-ipfs-api"
 	"github.com/sonr-io/sonr/x/schema/types"
 )
 
+const (
+	URL_PERSISTENCE_READ = "ipfs.Sonr.ws"
+)
+
+var (
+	ipfs_inter = shell.NewShell(URL_PERSISTENCE_READ)
+)
+
+func (k Keeper) LookUpContent(cid string, content interface{}) error {
+	out_path := filepath.Join(os.TempDir(), cid+".txt")
+	err := ipfs_inter.Get(cid, out_path)
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := os.ReadFile(out_path)
+
+	if err != nil {
+		return err
+	}
+
+	if err = json.Unmarshal(resp, &content); err != nil {
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(out_path)
+
+	return nil
+}
```

Comment:
Not in love with this downloading then reading the file, would prefer to do this in memory, but there are some tradeoffs there. Want thoughts on that

Reply:
Memory is better

Reply:
the interface doesnt supper memory buffers from the looks up it as its parameters are a hash (cid) and file path. so i dont think we can support an in memory model albeit unfortunate.

Reply:
updated to use `bufio` for in memory buffering of the file content a bit better, we still need to write to disk as the implantation of the underlying api is a bit weird with using tar ball unzips of the ipfs content.